apps/price_pusher/src/utils.ts

```diff
@@ -18,11 +18,11 @@ export function removeLeading0x(id: HexString): HexString {
   return id;
 }
 
-export function addLeading0x(id: HexString): HexString {
+export function addLeading0x(id: HexString): `0x${string}` {
   if (id.startsWith("0x")) {
-    return id;
+    return id as `0x${string}`;
   }
-  return "0x" + id;
+  return ("0x" + id) as `0x${string}`;
 }
```

Comment:
More idiomatic would be using a type guard:

```tsx
const addLeading0x = (id: HexString): `0x${string}` =>
    hasLeading0x(id) ? id : `0x${id}`;

const hasLeading0x = (input: string): input is `0x${string}` =>
    input.startsWith('0x');
```

Reply:
Thanks!